### PR TITLE
[IE TESTS] Move ExecGraph serialization tests from InferenceEngine functional tests to the shared library

### PR DESCRIPTION
**Pull request by iefode**
@ilyachur @jdanieck Please take a look

### CODE REVIEW

**Comment by ilyachur** [comment]
Thanks @iefode!
@jdanieck please check that these changes don't affect common serializer tests.

File: inference-engine/tests/functional/inference_engine/CMakeLists.txt [hunk 1]
```diff
@@ -217,6 +217,3 @@ else()
     #                                          PLUGIN_API)
 endif()
 
-
-# ir serialization functional tests variables
-target_compile_definitions(${TARGET_NAME} PRIVATE IR_SERIALIZATION_MODELS_PATH="${CMAKE_CURRENT_SOURCE_DIR}/ir_serialization/models/")
```
Comment (ilyachur) [inline inference-engine/tests/functional/inference_engine/CMakeLists.txt:222]: How are serializer tests working without this line?
As I understand you have moved only exec_graph tests to plugin specific place. 

And you also moved reference IRs. How are other serializer tests working without these models?  If you removed the definition of `IR_SERIALIZATION_MODELS_PATH`

Comment (jdanieck) [reply inference-engine/tests/functional/inference_engine/CMakeLists.txt:222]: As I understand variable with new model location is exposed via funcTestUtils cmake target (INTERFACE keyword) which is imported by ieFuncTests cmake target. @iefode is my reasoning correct?

`target_compile_definitions(${TARGET_NAME} INTERFACE IR_SERIALIZATION_MODELS_PATH="${CMAKE_CURRENT_SOURCE_DIR}/ir_serialization_models/")`

Comment (iefode) [reply inference-engine/tests/functional/inference_engine/CMakeLists.txt:222]: @jdanieck You point is correct.
And one more comment. Gays, when i was refactoring the test, i got a questing. As you know we decided to get rid of using IR-files. Is there another way to validate serialization functionality? (For example, create ngraph function -> serialize -> read serialized IR -> compare with source function) @ilyachur @jdanieck Is it possible? What do think?

Comment (ilyachur) [reply inference-engine/tests/functional/inference_engine/CMakeLists.txt:222]: If we have reference IRs it allows to compare with IR from the MO.

If we use the pipeline "create ngraph function -> serialize -> read serialized IR -> compare with source function" we don't have a check with IR, it means that serialized and MO IR formats can be different but all tests will be passed.

Comment (iefode) [reply inference-engine/tests/functional/inference_engine/CMakeLists.txt:222]: Ok, got it. Thanks

Comment (iefode) [reply inference-engine/tests/functional/inference_engine/CMakeLists.txt:222]: I decided to revert models moving because, cpu tests execute on different machine than ie build machine (and build, unit and ieFuncTests run on the same). And this makes no sense and test will fail :(

File: inference-engine/tests/functional/plugin/shared/src/execution_graph_tests/exec_graph_serialization.cpp [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    result << "ExpectedModelName=" << expected_model_name << "_";
+    result << "TargetDevice=" << targetDevice;
+    return result.str();
+}
+
+void ExecGraphSerializationTest::SetUp() {
+    const std::string XML_EXT = ".xml";
+    const std::string BIN_EXT = ".bin";
+
+    std::string test_name = ::testing::UnitTest::GetInstance()->current_test_info()->name();
```
[the remark this answers is not in the document]
Comment (iefode) [reply inference-engine/tests/functional/plugin/shared/src/execution_graph_tests/exec_graph_serialization.cpp:25]: Fixed, please check

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/execution_graph_tests/num_inputs_fusing_bin_conv.cpp [hunk 3]
```diff
@@ -5,7 +5,9 @@
 #include "execution_graph_tests/num_inputs_fusing_bin_conv.hpp"
 #include "common_test_utils/test_constants.hpp"
 
-using namespace LayerTestsDefinitions;
+namespace {
+using namespace ExecutionGraphTests;
 
 INSTANTIATE_TEST_CASE_P(smoke_inputsNumFusingBinConv, ExecGraphInputsFusingBinConv, ::testing::Values(CommonTestUtils::DEVICE_CPU),
                         ExecGraphInputsFusingBinConv::getTestCaseName);
+} // namespace
```
[the remark this answers is not in the document]
Comment (iefode) [reply inference-engine/tests/functional/plugin/cpu/shared_tests_instances/execution_graph_tests/num_inputs_fusing_bin_conv.cpp]: Done, thanks

File: inference-engine/tests/functional/plugin/shared/include/execution_graph_tests/keep_assing.hpp [hunk 4]
```diff
@@ -4,11 +4,11 @@
...
+namespace ExecutionGraphTests {
 
 class ExecGraphKeepAssignNode : public testing::TestWithParam<std::string> {
 public:
     static std::string getTestCaseName(testing::TestParamInfo<std::string> obj);
 };
 
-}  // namespace LayerTestsDefinitions
\ No newline at end of file
+}  // namespace ExecutionGraphTests
```
[the remark this answers is not in the document]
Comment (iefode) [reply inference-engine/tests/functional/plugin/shared/include/execution_graph_tests/keep_assing.hpp]: Done, thanks

File: inference-engine/tests/ie_test_utils/functional_test_utils/test_model/exec_graph_serialization_model.hpp [hunk 5]
```diff
@@ -0,0 +1,281 @@
+// Copyright (C) 2019 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <string>
+
+namespace FuncTestUtils {
+
+const char serialize_test_model[] = R"V0G0N(<?xml version="1.0" ?>
```
Comment (ilyachur) [inline inference-engine/tests/ie_test_utils/functional_test_utils/test_model/exec_graph_serialization_model.hpp:9]: I am not sure that we need to have these models in the header file.

Can we move these models to the test cpp file?

**Review by ilyachur — APPROVED** [review]

File: inference-engine/tests/ie_test_utils/functional_test_utils/CMakeLists.txt [hunk 6]
```diff
@@ -4,24 +4,33 @@
...
+        inference_engine_transformations
+        )
 
 addIeTarget(
     NAME ${TARGET_NAME}
     TYPE STATIC
     ROOT ${CMAKE_CURRENT_SOURCE_DIR}
     ADD_CPPLINT
     DEVELOPER_PACKAGE
-    LINK_LIBRARIES ngraphFunctions
```
Comment (ilyachur) [inline inference-engine/tests/ie_test_utils/functional_test_utils/CMakeLists.txt:16]: Why did you add `ngraphFunctions` library to public depenndencies?

Comment (iefode) [reply inference-engine/tests/ie_test_utils/functional_test_utils/CMakeLists.txt:16]: Good point, thanks, fixed)

File: inference-engine/tests/ie_test_utils/functional_test_utils/CMakeLists.txt [hunk 7]
```diff
@@ -4,24 +4,32 @@
...
-    EXPORT_DEPENDENCIES ${EXPORT_DEPENDENCIES}
+    LINK_LIBRARIES
+        PUBLIC
+            ${EXPORT_DEPENDENCIES}
+            ngraphFunctions
```
Comment (ilyachur) [inline inference-engine/tests/ie_test_utils/functional_test_utils/CMakeLists.txt:23]: The fix is not correct. You still add nGraphFunctions as a public dependency.

**Review by jdanieck — APPROVED** [review]

File: inference-engine/tests/ie_test_utils/functional_test_utils/CMakeLists.txt (hunk 7 above)
Comment (iefode) [reply inference-engine/tests/ie_test_utils/functional_test_utils/CMakeLists.txt:23]: I will review it in the next PR.
